plugins/@grouparoo/app-templates/src/source/table/profileProperties.ts

```diff
@@ -26,50 +29,55 @@ export const getProfileProperties: GetProfilePropertiesMethod = ({
     profiles,
     sourceOptions,
     sourceMapping,
-    property,
+    properties,
     propertyOptions,
     propertyFilters,
   }) => {
-    const responsesById: { [id: string]: any[] } = {};
+    const responsesById: ProfilePropertiesPluginMethodResponse = {};
     const tableName = sourceOptions[tableNameKey]?.toString();
-    const columnName = propertyOptions[columnNameKey]?.toString();
+    const columnNames = properties.map((p) =>
+      propertyOptions[p.id][columnNameKey]?.toString()
+    );
     const aggregationMethod = <AggregationMethod>(
-      propertyOptions[aggregationMethodKey]
+      propertyOptions[Object.keys(propertyOptions)[0]][aggregationMethodKey]
     );
-    const sortColumn = propertyOptions[sortColumnKey]?.toString();
+    const sortColumn =
+      propertyOptions[Object.keys(propertyOptions)[0]][
+        sortColumnKey
+      ]?.toString();
 
-    if (!aggregationMethod || !columnName) {
-      return undefined;
-    }
+    if (!aggregationMethod || columnNames.length === 0) return responsesById;
 
     const tableMappingCol: string = Object.values(sourceMapping)[0];
     const tablePrimaryKeyCol: string = Object.keys(sourceMapping)[0];
     const primaryKeysHash: { [pk: string]: string[] } = {};
 
     for (const i in profiles) {
-      const properties = await profiles[i].getProperties();
-      if (
-        properties[tableMappingCol]?.values.length > 0 &&
-        properties[tableMappingCol].values[0] // not null or undefined
-      ) {
-        const k = properties[tableMappingCol].values[0].toString();
-        if (!primaryKeysHash[k]) primaryKeysHash[k] = [];
-        primaryKeysHash[k].push(profiles[i].id);
-      }
-      if (
-        aggregationMethod === AggregationMethod.Sum ||
-        aggregationMethod === AggregationMethod.Count
-      ) {
-        //default all entries to 0 to start... will be replaced later if there is a response
-        responsesById[profiles[i].id] = [0];
+      for (const property of properties) {
+        const properties = await profiles[i].getProperties();
+        if (
+          properties[tableMappingCol]?.values.length > 0 &&
+          properties[tableMappingCol].values[0] // not null or undefined
+        ) {
+          const k = properties[tableMappingCol].values[0].toString();
+          if (!primaryKeysHash[k]) primaryKeysHash[k] = [];
+          primaryKeysHash[k].push(profiles[i].id);
+        }
+        if (
+          aggregationMethod === AggregationMethod.Sum ||
+          aggregationMethod === AggregationMethod.Count
+        ) {
+          //default all entries to 0 to start... will be replaced later if there is a response
+          responsesById[profiles[i].id] = { [property.id]: [0] };
```

Comment:
Looping through `properties`, this will set a new object for the profile every time that only contains whatever the latest property was. If the intention is to set all properties to `[0]` for that profile, shouldn't the object contain all properties instead of only the latest one?

Reply:
Very good catch!

Reply:
Update in 129ebe896cb4e252afee3f0c3de17df9ae58dfc3... but now I need to figure out why none of our existing tests had a problem with this.  It may be the case that since the Properties with these default values are not `"exact"`, so we never considered more than one property anyway...